### DISCUSSION TITLE
Static DAG definition and retry behavior

### DISCUSSION THREAD

**Discussion by JRoper18** [PrefectHQ/prefect#16309] [Q&A]
Hello,

Something I'm trying to do with prefect is to statically define a DAG of tasks and their dependencies, using the ".submit" and "wait_for" arguments. This works approximately well. One pattern I'm having trouble implementing is the ability to retry only failed tasks from a UI. For example, when a task fails, I may need to change some backend parameter or some configuration elsewhere. After that, I'd like to be able to resume running the DAG from the point where it failed. 

Is there a pattern for doing this?

**Comment by zzstoatzz** [comment]
hi @JRoper18 - this is a thing we're thinking a lot about as of late

you might be interested to read about our [plans](https://github.com/PrefectHQ/prefect/issues/16263) related to this.

Feel free to chime in there or follow up here!
